numpy/lib/format.py

```diff
@@ -259,6 +259,35 @@ def dtype_to_descr(dtype):
     else:
         return dtype.str
 
+def descr_without_paddings(descr):
+    aligned = False
+    newdescr = {'names': [], 'formats': [], 'offsets': []}
+    offset = 0
+    for tup in descr:
+        if tup[0]=='' and (tup[1].startswith('|V') or tup[1].startswith('V')) and isinstance(tup[1], str):
```

Comment:
I think it would be better to compute the field format first, and then skip if it's an unstructured void.

Reply:
adopted